File: Program's_Contributed_By_Contributors/Python_Programs/n-queens.py
class Solution:
    def solveNQueens(self, n: int) -> List[List[str]]:
        if n == 1:
            return [["Q"]]
        def display(board, arr):
            temp = []
            for row in board:
                st = ""
                for j in row:
                    if j:
                        st += "Q"
                        # print('Q',end= " ") 
                    else:
                        st += "."
                        # print('*', end = " ")
                # temp.append(st)
                # print(st)
                temp.append(st)
            # print("tempis ",temp)
            arr.append(temp)
            # return temp

        def isSafe(board, row, col):
            for i in range(row):
                if board[i][col]:
                    return False

            leftMost = min(row, col)
            for i in range(1, leftMost+1):
                if board[row-i][col-i]:
                    return False

            rightMost = min(row, len(board)-col-1)
            for i in range(1, rightMost+1):
                if board[row-i][col+i]:
                    return False

            return True

        def Queens(board, row, arr):
            if row == len(board):
                x = display(board, arr)
                print("")
                return x
            for col in range(len(board)):
                if isSafe(board, row, col):
                    board[row][col] = True
                    Queens(board, row+1, arr)
                    board[row][col] = False




        # main block
        board = [[False]*n for i in range(n)]
        arr = [0]
        Queens(board, 0, arr)
        arr.pop(0)
        return arr

File: Program's_Contributed_By_Contributors/C++ Programs/Array/Nqueens.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;
class Solution {
public:
    bool isSafe(vector<vector<int>>&board,int row, int col, int n)
    {
        for(int i=0;i<col;i++)
        {
            if(board[row][i]==1)
              return false;
        }
        for(int i=row,j=col;i>=0 && j>=0;i--,j--)
        {
            if(board[i][j]==1)
              return false;
        }
        for(int i=row,j=col;i<n && j>=0;i++,j--)
        {
            if(board[i][j]==1)
              return false;
        }
        return true;
    }
    void boardSolve(vector<vector<int>>&board, int col, int n, int &count)
    {
        int i;
        if(col>=n)
        {
            count+=1;
            return;
        }
        for(int i=0;i<n;i++)
        {
            if(isSafe(board,i,col,n))
            {
                board[i][col]=1;
                boardSolve(board,col+1,n,count);
                board[i][col]=0;
            }
        }
    }
    int totalNQueens(int n) 
    {
        vector<int>temp(n,0);
        vector<vector<int>>board(n,temp);
        int count=0;
        boardSolve(board,0,n,count);
        return count;
        
    }
    int main()
    {
        return totalNQueens(4);
    }
};
